opm/simulators/wells/WellState.cpp

```diff
@@ -724,6 +725,15 @@ void WellState<Scalar>::initWellStateMSWell(const std::vector<Well>& wells_ecl,
                     n_activeperf++;
                 }
             }
+            // If perf_data does not contain all active perfs, the well is distributed or inactive
+            // For inactive wells there is no need for calculating an initial solution
+            // \todo{ Update the procedure below to work for actually distributed wells. }
+            if (static_cast<int>(ws.perf_data.size()) != n_activeperf)
+                if (this->is_inactive_well(well_ecl.name()))
+                    continue;
```

Comment:
Can we use the following code, 
```c++
                if (this->is_inactive_well(well_ecl.name()))
                    continue;
```
right after the beginning of the for loop
![image](https://github.com/user-attachments/assets/7196bf67-e436-4d08-a211-675202800a9d)


for other scenarios, `if (static_cast<int>(ws.perf_data.size()) != n_activeperf)` we can throw while the new parallel ms wells development will need to handle it.